scripts/safe_template_linter.py

```diff
@@ -225,6 +262,48 @@ def __init__(self, rule, expression):
         self.end_line = 0
         self.end_column = 0
         self.lines = []
+        self.is_disabled = False
+
+    def _mark_expression_disabled(self, file_string, line_breaks):
+        """
+        Marks the expression violation as disabled if it finds the disable
+        pragma anywhere on either the first line of the violation, or the
+        preceding line of the violation.
+
+        Note that disable at the end of a line currently disables for the
+        following line as well.
+
+        Pragma format::
+
+            safe-lint: disable=violation-name,other-violation-name
+
+        Examples::
+
+            <% // safe-lint: disable=underscore-not-escaped %>
+            <%= gettext('Single Line') %>
+
+            <%= gettext('Single Line') %><% // safe-lint: disable=underscore-not-escaped %>
+
+        Arguments:
+            file_string: The contents of the file in violation.
+            line_breaks: A list of indices into file_string at which each line
+                break was found.
+
+        Side Effect:
+            Sets self.is_disabled as appropriate based on whether the pragma is
+            found.
+
+        """
+        start_line = 1 if self.start_line == 1 else self.start_line - 1
+        start_index = line_breaks[start_line - 1]
```

Comment:
Will these two lines work properly when `self.start_line` is 0? I think you may need to use `if self.start_line <= 1`.